geoh5py/data/text_data.py

```diff
@@ -18,15 +18,37 @@
 from __future__ import annotations
 
 import json
+from copy import deepcopy
 
 import numpy as np
+from numpy import ndarray
 
+from ..shared.utils import as_str_if_uuid, dict_mapper
 from .data import Data
 from .primitive_type_enum import PrimitiveTypeEnum
 
 
+def text_formating(values: None | np.ndarray | str) -> ndarray | None:
+    """
+    Format text values to utf-8.
+
+    :param values: The values to format.
+
+    :return: The formatted values.
+    """
+    values = deepcopy(values)
+
+    # todo: values[0] seems dangerous here
```

Comment:
Correct, we could do it previously because it would always come as an array. You probably will have to check if it's not a `str` below.